metrotranscom/doorway
# feat: update partner uploads

# Pull Request Template

## Issue Overview

This PR addresses the issue of uploads not working


## Description

The ability to upload files to S3 was only partially implemented due to some unexpected behavior on the Bloom side that made it impossible to complete via the previous direction.  This adds on to previous PRs to complete the work of implementing file uploads via the backend.

## How Can This Be Tested/Reviewed?

Both the backend and the partner site will need to be rebuilt and started up.  You will also need to have local AWS credentials for the uploads to actually go into S3.  For the backend site, update your `.env` with these values:

```
ASSET_FILE_SERVICE=s3
ASSET_FS_CONFIG_s3_REGION=us-west-1
ASSET_FS_CONFIG_s3_BUCKET=doorway-chriscasto-public-uploads20230601213649025400000006
ASSET_FS_CONFIG_s3_URL_FORMAT=public
```

Once both are running, log into the partner site.  Pick a listing (I used the one named "[doorway] Test: Default, No Preferences") and go down to Listing Photo.  Depending on the listing, there may be an image there already.  Click "Edit Photos"

![image](https://github.com/metrotranscom/doorway/assets/23032560/b6eafea7-985e-4861-b573-6d6d628e831c)

A box will show up where you can add a new image.  I used [this one](https://doorway-chriscasto-public-uploads20230601213649025400000006.s3.us-west-1.amazonaws.com/assets/756036/building/appartment-building-appartments-architecture-87223-1-1.jpg), but use whichever one you like under 5 MB.  The image should show up in the list.

![image](https://github.com/metrotranscom/doorway/assets/23032560/5e6c64a7-97ea-4ef4-b9e7-272a7c246b64)

Now go down to "Additional Eligibility Rules" and click on the button labelled "Add Building Selection Criteria".

![image](https://github.com/metrotranscom/doorway/assets/23032560/27d779fc-c5bb-499f-a945-01fa21b0bd3a)

A drawer will pop out with an option to upload a PDF or provide a URL.  Click on "Upload PDF".

![image](https://github.com/metrotranscom/doorway/assets/23032560/325ae2b6-0a51-421b-8798-198785ab403e)

Another file upload box will appear.  Upload a PDF (I used [this one](https://doorway-chriscasto-public-uploads20230601213649025400000006.s3.us-west-1.amazonaws.com/assets/159828/lottery/blank.pdf)).  

![image](https://github.com/metrotranscom/doorway/assets/23032560/d498515e-675c-4541-b5c9-bdc6588828b9)

Once the file has been uploaded, it should now show up for you under Building Selection Criteria.

![image](https://github.com/metrotranscom/doorway/assets/23032560/decbd92d-9670-4c90-b23c-cacdd19472c2)

Save the listing to commit the changes to the database, then reload to verify the changes have persisted.

Finally, upload lottery results.  Not all listings are set up for this, but I was able to do so under the listing named "[doorway] Test: Default, Lottery Results Pending".  If the listing is eligible, you should see a "Post Results" option when in edit mode.

![image](https://github.com/metrotranscom/doorway/assets/23032560/3b553543-f832-4f23-8613-315f385564d5)

Click on that, then upload a PDF file as before.  Click on "Post".

![image](https://github.com/metrotranscom/doorway/assets/23032560/335602a4-4d48-4c78-a3cb-2738d8269eb0)

If you scroll down to the bottom of the page, you should see a "Preview Lottery Results" link that points to the file you just uploaded.

![image](https://github.com/metrotranscom/doorway/assets/23032560/787d7267-f452-4852-a7de-00052994d942)


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
